lib/benchpark/cuda.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict
+import yaml  # TODO: some way to ensure yaml available
+
+from benchpark.directives import ExperimentSystemBase
+import benchpark.spec
+import benchpark.paths
+import benchpark.repo
+import benchpark.runtime
+import benchpark.variant
+
+bootstrapper = benchpark.runtime.RuntimeResources(benchpark.paths.benchpark_home)
+bootstrapper.bootstrap()
+
+import ramble.language.language_base  # noqa
+import ramble.language.language_helpers  # noqa
+
+
+class CudaExperiment(ExperimentSystemBase):
+    """This is the superclass for all benchpark experiments.
+
+    ***The Experiment class***
+
+    Experiments are written in pure Python.
+
+    There are two main parts of a Benchpark experiment:
+
+      1. **The experiment class**.  Classes contain ``directives``, which are
+         special functions, that add metadata (variants) to packages (see
+         ``directives.py``).
+
+      2. **Experiment instances**. Once instantiated, an experiment is
+         essentially a collection of files defining an experiment in a
+         Ramble workspace.
+    """
+
+    variant("cuda", default=False, description="Build and run with CUDA")
+
+
+
+    #
+    # These are default values for instance variables.
+    #
+
+    # This allows analysis tools to correctly interpret the class attributes.
+    variants: Dict[
+        "benchpark.spec.Spec",
+        Dict[str, benchpark.variant.Variant],
+    ]
```

Comment:
You should not have to instantiate this here. Why does this inherit from `ExperimentSystemBase` vs. `Experiment`? If this is related to MRO concerns, see my later comment about using composition.